src/server/frontend/wayland/wl_mir_window.cpp

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright © 2015-2018 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: Christopher James Halse Rogers <christopher.halse.rogers@canonical.com>
+ */
+
+#include "wl_mir_window.h"
+
+#include "wayland_connector.h"
+#include "wl_surface.h"
+#include "basic_surface_event_sink.h"
+
+#include "mir/frontend/shell.h"
+#include "mir/frontend/session.h"
+#include "mir/frontend/event_sink.h"
+
+#include "mir/scene/surface.h"
+#include "mir/scene/surface_creation_parameters.h"
+
+namespace mir
+{
+namespace frontend
+{
```

Comment:
I know this is copied from wayland_connector.cpp, but we don't open namespace scopes for the bulk of implementation files. we prefer:
````
using mf = mir::frontend;
...

mf::WlMirWindow* const 
mf::null_wl_mir_window_ptr = &null_wl_mir_window_instance
````

Consider, if you misspell "null_wl_mir_window_ptr" the suggested version will give a warning at compile (or even edit) time, your version fails at link time.

(I won't block on this as it is "pre-existing" code)